packages/auth-password/index.js

```diff
@@ -95,8 +96,20 @@ class PasswordAuthStrategy {
 
   async _matchItem(item, args, secretFieldInstance) {
     const { secretField } = this.config;
+    let hash = item[secretField];
     const secret = args[secretField];
-    return await secretFieldInstance.compare(secret, item[secretField]);
+    let message = `[passwordAuth:secret:mismatch] The ${secretField} provided is incorrect`;
+    if (!hash) {
+      hash = await secretFieldInstance.generateHash('password1234');
+      await secretFieldInstance.compare(secret, hash);
+      return {
+        success: false,
```

Comment:
If we are returning false here why are we bothering checking the `secretFieldInstance` above?

Reply:
this was due to comment earlier that we should have `work factor` factored in for this. comparing password would keep the system busy for that time and guess work for time will not be easy.

We are returning only the password incorrect error when `protectIdentities = true` this means we have to use the cpu cycle before returning false.

Reply:
I guess my understanding of work factor is something different. I thought this was just a value given to the hashing algorithm. I still don't understand why we are doing unnecessary checks for the sake of increasing CPU cycles. Is the intent to slow down automated brute force attacks?

Reply:
yes, some existing comments from file.

```js
   // If we failed to match an identity and we're protecting existing identities then combat timing
    // attacks by creating an arbitrary hash (should take about as long has comparing an existing one)
    if (results.length !== 1 && this.config.protectIdentities) {
      // TODO: This should call `secretFieldInstance.compare()` to ensure it's
      // always consistent.
      // This may still leak if the workfactor for the password field has changed
```

we may have to revisit in future to add unnecessary delay in the response for every wrong password attempt from same client, or even option to integrate re-captcha to password authentication